### ISSUE TITLE
Pass 'read_dictionary' kwarg to pyarrow

### ISSUE THREAD

**Issue by sapientderek** [awslabs/aws-data-wrangler#160]
Thanks for the great package! The Athena CTAS method is much faster than more robust than other packages/methods.

It appears the underlying pyarrow package memory usage spikes when using `wr.s3.read_parquet()` and since it uses it, `wr.athena.read_sql_query()`.

With 16 GB RAM, my python kernel is crashing due to a memory error (checking `free` in separate shell during execution shows this).

Looking for work arounds I found pyarrow has the kwarg `read_dicitonary` in pyarrow.parquet.ParquetDataset which reduces memory use and speeds up read by directly reading string columns as categories. Pandas's `read_parquet` passes kwargs to the underlying engine (pyarrow). 

To check this I downloaded the parquet files the Athena query generated:
```
Line #    Mem usage    Increment   Line Contents
================================================
     4     76.1 MiB     76.1 MiB   @profile
     5                             def read_parquet_files(files):
     6     76.1 MiB      0.0 MiB       dfs=[]
     7   5248.9 MiB      0.0 MiB       for file in files:
     8   5248.9 MiB   3701.6 MiB           df = pd.read_parquet(file,read_dictionary=['a','b'])
     9   5248.9 MiB      0.0 MiB           dfs.append(df)
    10   4694.2 MiB      0.0 MiB       total_df = pd.concat(dfs,ignore_index=True,copy=False)

real	0m15.570s
user	0m11.107s
sys	0m7.833s
```
Ultimately this leaves the categorized columns as strings. They can be kept as categories, and further reducing memory and improving speed by using `pandas.api.types.union_categoricals()`

```
Line #    Mem usage    Increment   Line Contents
================================================
     5     76.5 MiB     76.5 MiB   @profile
     6                             def read_parquet_files(files):
     7     76.5 MiB      0.0 MiB       dfs = []
     8     76.5 MiB      0.0 MiB       a_categories = []
     9     76.5 MiB      0.0 MiB       b_categories = []
    10   5279.5 MiB      0.0 MiB       for file in files:
    11   5279.5 MiB   3800.8 MiB           df = pd.read_parquet(file,read_dictionary=['a','b'])
    12                                     
    13   5279.5 MiB      0.0 MiB           a_categories.append(df['a'])
    14   5279.5 MiB      0.0 MiB           b_categories.append(df['b'])
    15                                     
    16   5279.5 MiB      0.0 MiB           dfs.append(df)
    17   2885.0 MiB      0.0 MiB      total_a_categories = union_categoricals(a_cats).categories
    18   2046.0 MiB      0.0 MiB       total_b_categories = union_categoricals(b_cats).categories
    19   2046.0 MiB      0.0 MiB       for i in range(len(dfs)):
    20   2046.0 MiB      0.0 MiB           dfs[i]['a'].cat.set_categories(total_a_categories,inplace=True)
    21   2046.0 MiB      0.0 MiB           dfs[i]['b'].cat.set_categories(total_b_categories,inplace=True)
    22   3538.4 MiB   1492.5 MiB       total_df = pd.concat(dfs,ignore_index=True,copy=False)



real	0m6.288s
user	0m8.040s
sys	0m4.487s
```

A possible simple enhancement to awswrangler would being able to pass kwargs to pandas/pyarrow.parquet.ParquetDataset.

A more involved enhancement would be combining categoricals when `read_dictionary` is not None.

**Comment by igorborgest** [comment]
@sapientderek thanks! Really cool analyses!!! 👏👏👏

I'm messing around with your suggestion, let's see what I will be able to adapt on the next release.

**Comment by igorborgest** [comment]
@sapientderek feature available on version [1.0.1](https://github.com/awslabs/aws-data-wrangler/releases/tag/1.0.1). Thanks!
